sonar-scanner-engine/src/main/java/org/sonar/scanner/report/ContextPropertiesPublisher.java

```diff
@@ -20,31 +20,45 @@
 package org.sonar.scanner.report;
 
 import com.google.common.base.Function;
+import com.google.common.collect.Iterables;
 import java.util.Map;
+import javax.annotation.Nonnull;
+import org.sonar.api.config.Settings;
 import org.sonar.scanner.protocol.output.ScannerReport;
 import org.sonar.scanner.protocol.output.ScannerReportWriter;
 import org.sonar.scanner.repository.ContextPropertiesCache;
 
 import static com.google.common.collect.FluentIterable.from;
 
 public class ContextPropertiesPublisher implements ReportPublisherStep {
+
   private final ContextPropertiesCache cache;
+  private final Settings settings;
 
-  public ContextPropertiesPublisher(ContextPropertiesCache cache) {
+  public ContextPropertiesPublisher(ContextPropertiesCache cache, Settings settings) {
     this.cache = cache;
+    this.settings = settings;
   }
 
   @Override
   public void publish(ScannerReportWriter writer) {
-    Iterable<ScannerReport.ContextProperty> it = from(cache.getAll().entrySet()).transform(new MapEntryToContextPropertyFunction());
-    writer.writeContextProperties(it);
+    MapEntryToContextPropertyFunction transformer = new MapEntryToContextPropertyFunction();
+
+    Iterable<ScannerReport.ContextProperty> fromCache = from(cache.getAll().entrySet())
+      .transform(transformer);
+
+    Iterable<ScannerReport.ContextProperty> fromSettings = from(settings.getProperties().entrySet())
+      .filter(e -> e.getKey().startsWith("sonar.analysis."))
```

Comment:
that's a magic number here, a constant with a descriptive name would be good, maybe shared with the same magic number in `WebhookPayload.java`